int16 num_waypoints
Waypoint[] waypoints